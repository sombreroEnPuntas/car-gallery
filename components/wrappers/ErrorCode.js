import styled from 'styled-components'
import { colors, fonts } from '.'

const ErrorCode = styled.code`
  color: ${colors.error.outline};
  font-family: ${fonts.fontFamily};
  font-size: 0.65em;
  font-weight: bold;
`

export default ErrorCode
