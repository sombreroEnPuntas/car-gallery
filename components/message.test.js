// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Dependencies
import ErrorCode from './wrappers/ErrorCode'

// Tested Unit
import TestedComponent from './message'

const getProps = customProps => ({
  message: 'Hello world!',
  ...customProps,
})

describe('Message', () => {
  const wrapper = shallow(<TestedComponent {...getProps()} />)

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('shows "Hello world!"', () => {
    const actual = wrapper.find(ErrorCode).text()
    const expected = 'Hello world!'

    expect(actual).toEqual(expected)
  })

  it('shows Loading status', () => {
    wrapper.setProps({ message: null })

    const actual = wrapper.find('h3').text()
    const expected = 'Loading '

    expect(actual).toEqual(expected)
  })
})
