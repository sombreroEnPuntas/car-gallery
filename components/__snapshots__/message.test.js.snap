// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message matches the snapshot 1`] = `
<MessageBox>
  <h3>
    Ooops...
  </h3>
  <p>
    Something went sideways :$
  </p>
  <ErrorBox>
    <ErrorCode>
      Hello world!
    </ErrorCode>
  </ErrorBox>
  <p>
    Try again later!
  </p>
</MessageBox>
`;
